src/MoBi.Presentation/Presenter/AddContentToModulePresenter.cs

```diff
@@ -23,7 +23,7 @@ protected AddContentToModulePresenter(TView view) : base(view)
 
       public void AddMoleculesSelectionChanged(bool moleculesSelected)
       {
-         if (!moleculesSelected || Module.InitialConditionsCollection.Any())
+         if (!moleculesSelected || (Module?.InitialConditionsCollection != null && Module.InitialConditionsCollection.Any()))
```

Comment:
I think there is an earlier issue than this. By the time this get's called, the presenter should already be initialized and that initialization should always include the module to which we are adding initial conditions.

Can you check what happens when you left click, then select "Add Building Block" the pick "Initial Conditions" I think the Module will be non-null.

Reply:
Yeah, I`ve made some analysis before submitting.
As you mention, the only place where we initialize this variable is on the "AddBuildingBlocksToModule" method:
![image](https://github.com/user-attachments/assets/d91ab90d-d10f-44e7-bc56-e73e833cbd0e)

Other than that, the Module property is null.
What I think (might be mistaken of course), is that this module variable is being set only on this public method, instead of the private method (with the same name) for a reason, but checking references, I can see that this property is only being used for this case mentioned on the ticket, so it might be safe to move it into the private method.

Let me know your thoughts on this.